mapie/control_risk/crc_rcps.py

```diff
@@ -0,0 +1,211 @@
+        if bound == "hoeffding":
+            r_hat_plus = (
+                r_hat +
+                np.sqrt((1 / (2 * n_obs)) * np.log(1 / delta))
+            )
+
+        elif bound == "bernstein":
+            sigma_hat_bern = np.var(r_hat, axis=0, ddof=1)
+            r_hat_plus = (
+                r_hat +
+                np.sqrt((sigma_hat_bern * 2 * np.log(2 / delta)) / n_obs) +
+                (7 * np.log(2 / delta)) / (3 * (n_obs - 1))
+            )
+
+        else:
+            mu_hat = (
+                (.5 + np.cumsum(risks, axis=0)) /
+                (np.repeat(
+                    [range(1, n_obs + 1)],
+                    n_lambdas,
+                    axis=0
+                ).T + 1)
+            )
+            sigma_hat = (
+                (.25 + np.cumsum((risks - mu_hat)**2, axis=0)) /
+                (np.repeat(
+                    [range(1, n_obs + 1)],
+                    n_lambdas,
+                    axis=0
+                ).T + 1)
+            )
+            sigma_hat = np.concatenate(
+                [
+                    np.full(
+                        (1, n_lambdas), fill_value=sigma_init
+                    ), sigma_hat[:-1]
+                ]
+            )
+            nu = np.minimum(
+                1,
+                np.sqrt((2 * np.log(1 / delta)) / (n_obs * sigma_hat))
+            )
+
+            # Split the calculation in two to prevent memory issues
+            batches = [
+                range(int(n_obs / 2)),
+                range(n_obs - int(n_obs / 2), n_obs)
+            ]
+            K_R_max = np.zeros((n_lambdas, n_lambdas))
+            for batch in batches:
+                nu_batch = nu[batch]
+                losses_batch = risks[batch]
+
+                nu_batch = np.repeat(
+                    np.expand_dims(nu_batch, axis=2),
+                    n_lambdas,
+                    axis=2
+                )
+                losses_batch = np.repeat(
+                    np.expand_dims(losses_batch, axis=2),
+                    n_lambdas,
+                    axis=2
+                )
+
+                R = lambdas
+                K_R = np.cumsum(
+                    np.log(
+                        (
+                            1 -
+                            nu_batch *
+                            (losses_batch - R)
+                        ) +
+                        np.finfo(np.float64).eps
+                    ),
+                    axis=0
+                )
+                K_R = np.max(K_R, axis=0)
+                K_R_max += K_R
+
+            r_hat_plus_tronc = lambdas[np.argwhere(
+                np.cumsum(K_R_max > -np.log(delta), axis=1) == 1
+            )[:, 1]]
+            r_hat_plus = np.ones(n_lambdas)
+            r_hat_plus[:len(r_hat_plus_tronc)] = r_hat_plus_tronc
+
+    else:
+        r_hat_plus = (n_obs / (n_obs + 1)) * r_hat + (1 / (n_obs + 1))
+
+    return r_hat, r_hat_plus
+
+
+def _find_lambda_star(
```

Comment:
SUGGESTION: remove the `_` character.
```suggestion
def find_lambda_star(
```
Adapt this change in all files.

Reply:
Done